lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -3125,6 +3255,36 @@ void PullbackCloner::Implementation::visitSILBasicBlock(SILBasicBlock *bb) {
         break;
       }
       }
+    } else if (auto *tai = dyn_cast<TryApplyInst>(bbArg->getSingleTerminator())) {
+      // try_apply does not provide result, so there is no value to propagate
+      // adjoint to.  Prepare adjoint and associate it "on side", so it will be
+      // used as seed during pullback call emission
+      if (!getPullbackInfo().shouldDifferentiateApplySite(tai))
```

Comment:
It looks like that existing tests do not cover this condition and it always evaluates to false - inserting `assert(getPullbackInfo().shouldDifferentiateApplySite(tai);` does not break any tests.